src/Npgsql/HardCancellationBlock.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Threading;
+
+namespace Npgsql
+{
+    internal class HardCancellationBlock : IDisposable
+    {
+        bool disposed;
```

Comment:
Is it possible that one block may be nested?

Reply:
It actually is. There is `GetFieldValueSequential` method, which calls `SeekToColumnSequential`. And while in that instance it's possible to rewrite so there wouldn't be, I think allowing nested blocks is a better solution.